### PR TITLE
[WIP] Add jscs javascript validation

### PR DESCRIPTION
**Pull request by masakura**
JSHint no longer report the inspection of indent. Validation that overlap with JSCS an JSHint's option seems to go been deleted.

http://jshint.com/docs/options/#indent

> Warning This option has been deprecated and will be removed in the next major release of JSHint. JSHint is limiting its scope to issues of code correctness. If you would like to enforce rules relating to code style, check out the JSCS project.

Would you add JSCS validation, not only JSHint?

[JSCS](https://github.com/jscs-dev/node-jscs) and [grunt-jscs](https://github.com/jscs-dev/grunt-jscs)

### CODE REVIEW

**Comment by addyosmani** [comment]
Commenting on the feature: I personally don't mind JSCS validation being included as part of the build. I do this in some of my personal projects. That said, @silvenon may have better insight on whether this is something our users have been asking about.

**Comment by silvenon** [comment]
There haven't been many requests for this, I recall only one followed by some :+1:s, but as JSHint is deprecating its options and moving them to JSCS, I see less and less reason not to implement it.

So :+1: from me too.

**Comment by addyosmani** [comment]
Btw, this change would benefit from rebasing against master.

**Comment by masakura** [comment]
I'm sorry...

I was merged it by mistale...

**Comment by silvenon** [comment]
What do you mean? Looks to me like you merged master in to bring your code up to date? You can squash your commits into one :wink:

**Comment by masakura** [comment]
Squash and rebase completed. Is this all right?

File: app/templates/Gruntfile.js [hunk 1]
```diff
@@ -424,6 +433,7 @@ module.exports = function (grunt) {
     }
   });
 
+  grunt.registerTask('jscheck', ['jshint', 'jscs']);
```
Comment (stevemao) [inline app/templates/Gruntfile.js]: a better name would be `lint`?

**Comment by stevemao** [comment]
Why not adding jscs validating this project itself?

**Comment by dvbeato** [comment]
I would like to have JSCS too.

File: app/templates/Gruntfile.js (hunk 1 above)
Comment (eddiemonge) [inline app/templates/Gruntfile.js]: Im good with `jschecks`

File: app/templates/Gruntfile.js [hunk 2]
```diff
@@ -24,6 +24,16 @@ module.exports = function (grunt) {
     dist: 'dist'
   };
 
+  config.targets = {
+    // Validate target javascripts
+    jscheck: [
+      'Gruntfile.js',
+      '<%%= config.app %>/scripts/{,*/}*.js',
+      '!<%%= config.app %>/scripts/vendor/*',
```
Comment (eddiemonge) [inline app/templates/Gruntfile.js]: you can delete this line since it only exists in config.dist

File: app/templates/Gruntfile.js [hunk 3]
```diff
@@ -480,6 +490,7 @@ module.exports = function (grunt) {
 
   grunt.registerTask('default', [
     'newer:jshint',
+    'newer:jscs',
```
Comment (eddiemonge) [inline app/templates/Gruntfile.js]: `newer:jschecks` no?

File: app/templates/Gruntfile.js (hunk 1 above)
Comment (stevemao) [inline app/templates/Gruntfile.js]: `check` could mean anything but `lint`means "a particular program that flagged some suspicious and non-portable constructs (likely to be bugs)". Both `jshint` and `jscs` are linting software

https://github.com/gulpjs/gulp/blob/master/package.json#L59

Comment (masakura) [inline app/templates/Gruntfile.js]: Hi! @stevemao 
- `lint` - not validate css/html
- `jslint` - same name linter exists

What about `js-lints`?

File: app/templates/Gruntfile.js (hunk 2 above)
Comment (masakura) [inline app/templates/Gruntfile.js]: Hi! @eddiemonge 

Merge config.targs to config. OK? e61e2f3

File: app/templates/Gruntfile.js (hunk 1 above)
Comment (stevemao) [inline app/templates/Gruntfile.js]: its just my suggestion as `gulp` is using `lint`. Would you validate css/html?

Comment (eddiemonge) [inline app/templates/Gruntfile.js]: I would definitely validate HTML if I was still writing it by hand. Same for CSS.

**Comment by arthurvr** [comment]
Also needs a fix for the merge conflict.

**Comment by arthurvr** [comment]
ping @masakura

**Comment by silvenon** [comment]
Does anyone know how are ESLint and JSCS connected? Can they work together or is there an overlap?

**Comment by silvenon** [comment]
Oh, and I'll rebase this onto master one of these days.

**Comment by stevemao** [comment]
They don't need to work together. ESLint also checks styles. It's probably why I would choose eslint as I don't want to install two tools (assume they all equally good)

**Comment by arthurvr** [comment]
At generator-gulp-webapp we switched to ESLint so I'd like to be consistent in this repo.

Replacing our current JSHint config with ESLint would make this PR needless (thanks for the hard work, though, @masakura!)

> Does anyone know how are ESLint and JSCS connected? Can they work together or is there an overlap?

As @stevemao said. ESLint does all the things we need.

**Comment by silvenon** [comment]
Ok, we'll switch to ESLint then. Thanks for your work @masakura! I really hate turning down a PR, but I only found about ESLint recently :confused:

**Pull request closed without merging**